students/donghwa/users/views.py

```diff
@@ -1,3 +1,44 @@
 from django.shortcuts import render
 
 # Create your views here.
+
+import json, re
+
+from django.http     import JsonResponse
+from django.views    import View
+
+from users.models    import User
+
+class SignUpView(View):
+    def post(self, request):
+
+        try:
+            data = json.loads(request.body)
+
+
+            regexp_email    = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+            regexp_password = re.compile('^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$')
+
+            if re.match(regexp_email, data['email']) is None:
+                return JsonResponse({'message' : 'WRONG_EMAIL_FORMAT'}, status=400)
+
+            if re.match(regexp_password, data['password']) is None:
+                return JsonResponse({'message' : 'WRONG_PASSWORD_FORMAT'}, status=400)
+
+            if User.objects.filter(email = data['email']).exists():
+                return JsonResponse({'message' : 'USER_ALREADY_EXISTS'}, status=400)
+
+            if re.match(regexp_email, data['email']) and re.match(regexp_password, data['password']):
+
+                 User.objects.create(
+                     name        = data['name'],
+                     email       = data['email'],
+                     password    = data['password'],
+                     contacts    = data['contacts'],
+                     address     = data['address']
+                 )
```

Comment:
이미 위에 작성하신 `if 문`을 통해서 정규표현식을 다 거쳐왔기 때문에 다시 한번 검증할 필요가 없을 것 같습니다 동화님!
```suggestion
                 User.objects.create(
                     name        = data['name'],
                     email       = data['email'],
                     password    = data['password'],
                     contacts    = data['contacts'],
                     address     = data['address']
                 )
```